ga4gh/datamodel/variants.py

```diff
@@ -342,3 +343,118 @@ def getMetadata(self):
         # TODO: Implement this
         ret = []
         return ret
+
+
+def _encode_value(value):
+    if isinstance(value, (list, tuple)):
+        return [str(v) for v in value]
+    else:
+        return [str(value)]
+
+
+_nothing = object()
+
+
+def is_empty_iter(it):
+    """Return True iff the iterator is empty or exhausted"""
+    return next(it, _nothing) is _nothing
+
+
+class HtslibVariantSet(VariantSet):
+    """
+    Class representing a single variant set backed by a directory of indexed
+    VCF or BCF files.
+    """
+
+    def __init__(self, variantSetId, vcfPath, use_bcf=True):
+        self._variantSetId = variantSetId
+        self._created = protocol.convertDatetime(datetime.datetime.now())
+        self._chromFileMap = {}
+
+        for pattern in ['*.bcf', '*.vcf.gz']:
+            for filename in glob.glob(os.path.join(vcfPath, pattern)):
+                self._addFile(filename)
+
+    def _addFile(self, filename):
+        varFile = pysam.VariantFile(filename)
+        if not varFile.index:
+            raise Exception("VCF/BCF files must be indexed")
+        for chrom in varFile.index:
+            # Unlike Tabix indices, CSI indices include all contigs defined
+            # in the BCF header.  Thus we must test each one to see if
+            # records exist or else they are likely to trigger spurious
+            # overlapping errors.
+            if is_empty_iter(varFile.fetch(chrom)):
+                continue
+            if chrom in self._chromFileMap:
+                raise Exception("cannot have overlapping VCF/BCF files.")
+            self._chromFileMap[chrom] = varFile
+
+    def convertVariant(self, record):
+        variant = protocol.GAVariant()
```

Comment:
I don't think we need all the blank lines in this method --- these are all very simple statements, and don't need to be separated.